wsd/wopi/WopiStorage.cpp

```diff
@@ -423,10 +423,123 @@ StorageBase::LockUpdateResult WopiStorage::updateLockState(const Authorization&
     return LockUpdateResult(LockUpdateResult::Status::FAILED, failureReason);
 }
 
-void WopiStorage::updateLockStateAsync(const Authorization& /*auth*/, LockContext& /*lockCtx*/,
-                                       LockState /*lock*/, const Attributes& /*attribs*/,
-                                       const AsyncLockStateCallback& /*asyncLockStateCallback*/)
+void WopiStorage::updateLockStateAsync(const Authorization& auth, LockContext& lockCtx,
+                                       LockState lock, const Attributes& attribs,
+                                       SocketPoll& socketPoll,
+                                       const AsyncLockStateCallback& asyncLockStateCallback)
 {
+    auto profileZone = std::make_shared<ProfileZone>(
```

Comment:
I have not read the WOPI spec thoroughly, hence I can't properly review this yet. 
Other that this - the async change and its pattern looks good!

In general, as discussed, I like us to add the used section of the specification in the header API doc, allowing us to quickly validated and also know the impact etc.

Reply:
Thanks. Understood. Although this code doesn't make any WOPI changes (it's changing an existing synchronous request to be async), more documentation is good.